src/Streamly/Fold.hs

```diff
@@ -952,14 +950,21 @@ ltakeWhile predicate (Fold step initial done) = Fold step' initial' done'
 -- prefix of length @n@ and the second group consisting of the rest of the
 -- stream.
 --
+-}
+{-# INLINE splitAt #-}
 splitAt
     :: Monad m
     => Int
     -> Fold m a b
     -> Fold m a c
     -> Fold m a (b, c)
-splitAt n = undefined
--}
+splitAt n (Fold stepL initL doneL) (Fold stepR initR doneR) = Fold step init done
+  where
+    step (ind,v1,v2) input =
+            if ind > 0 then stepL v1 input >>= (\a -> return (ind-1,a,v2))
+                       else stepR v2 input >>= (\b -> return (ind,v1,b))
+    init  = (,,) <$> return n <*> initL <*> initR
```

Comment:
1) Move the `init` function before `step` function, that's the logical order to read the code.

2) Use `Pair3'` from `Streamly.Strict` instead of using `(,,)`. Always use a strict data type for the left fold accumulator.